### PR TITLE
Course details

### PR DESCRIPTION
**Pull request by Jeyadheesh**

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -74,6 +60,16 @@ function App() {
             path="/admin/generate"
           />
 
+          {/* Feedback Homepage */}
```
Comment (RoshanKumar28) [inline client/src/App.tsx:63]: protected route ah yen remove panne?

File: client/src/components/Header.tsx [hunk 2]
```diff
@@ -1,18 +1,42 @@
...
 import Button from "./Button";
+import useRole from "../store/useRole";
+import useUserDetails from "../store/useUserDetails";
+import { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
 
 type Props = {};
 
 const Header = (props: Props) => {
+  const { role, setRole } = useRole();
```
Comment (RoshanKumar28) [inline client/src/components/Header.tsx:12]: logout session ah correct ah destroy pannutha?

File: client/src/components/ProtectedRoute.tsx [hunk 3]
```diff
@@ -10,25 +10,25 @@ type Props = {
 const ProtectedRoute = ({ children, shouldBeAdmin = false }: Props) => {
   const role = useRole((state) => state.role);
 
-  if (!role) {
-    return <Navigate to={"/"} />;
-  }
-
+  // if (!role) {
```
Comment (RoshanKumar28) [inline client/src/components/ProtectedRoute.tsx:13]: yen full ah comment pani vechurke?

File: server/app.js [hunk 4]
```diff
@@ -258,7 +258,7 @@ app.post("/loginAuth", (req, res) => {
         }
         if (result[0]) {
           const token = jwt.sign(
```
Comment (RoshanKumar28) [inline server/app.js:260]: role vechu matum signIn panna pathatha... username vechu yenda sign panre

Comment (Jeyadheesh) [reply server/app.js:260]: ipo role user aa irukkum podhu .. avaen login ku ponna ... feedback/{username} nu thana redirect pannanum .. apo username aa session irundhu mattum tha edukka mudiyum ... refresh pannalum panniralam

File: client/src/components/ProtectedRoute.tsx (hunk 3 above)
Comment (Jeyadheesh) [reply client/src/components/ProtectedRoute.tsx:13]: /me role aa change pandradhukku munnadiyae .. idhu role = null  aa padhu redirect pannirudhu ...

File: server/app.js (hunk 4 above)
Comment (RoshanKumar28) [reply server/app.js:260]: ohh ryt uh

File: client/src/components/Header.tsx (hunk 2 above)
Comment (Jeyadheesh) [reply client/src/components/Header.tsx:12]: work agudhu da

File: client/src/App.tsx (hunk 1 above)
Comment (Jeyadheesh) [reply client/src/App.tsx:63]: protected oda pannalum crt than da varudhu ...  protected vachuraen. .

**Pull request merged**